docs/develop/test_infrastructure.md

```diff
@@ -0,0 +1,17 @@
+# Testing Infrastructure
+
+Where possible, additions to the code should be carried out in a
+[test-driven development](https://en.wikipedia.org/wiki/Test-driven_development)
+manner:
+
+> **Write failing tests that the code should pass, then write code to pass the tests**.
```

Comment:
You know this could be a "warn" directive now, right ;-)

Reply:
Well its meant to be emphasized more than a warning :)